build.cake

```diff
@@ -159,56 +159,29 @@ Task("__RunTests")
     }
 });
 
-Task("__RunMutationTests")
+Task("__RunCoreMutationTests")
     .Does((context) =>
 {
-    var runMutationTests = EnvironmentVariable("RUN_MUTATION_TESTS") switch
-    {
-        "false" => false,
-        _ => true
-    };
-
-    if (!runMutationTests)
-    {
-        return;
-    }
-
     var oldDirectory = context.Environment.WorkingDirectory;
     context.Environment.WorkingDirectory = MakeAbsolute(Directory("test"));
 
-    TestProject(File("../src/Polly.Core/Polly.Core.csproj"), File("./Polly.Core.Tests/Polly.Core.Tests.csproj"), "Polly.Core.csproj");
-    TestProject(File("../src/Polly.RateLimiting/Polly.RateLimiting.csproj"), File("./Polly.RateLimiting.Tests/Polly.RateLimiting.Tests.csproj"), "Polly.RateLimiting.csproj");
-    TestProject(File("../src/Polly.Extensions/Polly.Extensions.csproj"), File("./Polly.Extensions.Tests/Polly.Extensions.Tests.csproj"), "Polly.Extensions.csproj");
-    TestProject(File("../src/Polly.Testing/Polly.Testing.csproj"), File("./Polly.Testing.Tests/Polly.Testing.Tests.csproj"), "Polly.Testing.csproj");
-    TestProject(File("../src/Polly/Polly.csproj"), File("./Polly.Specs/Polly.Specs.csproj"), "Polly.csproj");
+    MutationTestProject(File("../src/Polly.Core/Polly.Core.csproj"), File("./Polly.Core.Tests/Polly.Core.Tests.csproj"), "Polly.Core.csproj");
+    MutationTestProject(File("../src/Polly.RateLimiting/Polly.RateLimiting.csproj"), File("./Polly.RateLimiting.Tests/Polly.RateLimiting.Tests.csproj"), "Polly.RateLimiting.csproj");
+    MutationTestProject(File("../src/Polly.Extensions/Polly.Extensions.csproj"), File("./Polly.Extensions.Tests/Polly.Extensions.Tests.csproj"), "Polly.Extensions.csproj");
+    MutationTestProject(File("../src/Polly.Testing/Polly.Testing.csproj"), File("./Polly.Testing.Tests/Polly.Testing.Tests.csproj"), "Polly.Testing.csproj");
 
     context.Environment.WorkingDirectory = oldDirectory;
+});
 
-    void TestProject(FilePath proj, FilePath testProj, string project)
-    {
-        var dotNetBuildSettings = new DotNetBuildSettings
-        {
-            Configuration = "Debug",
-            Verbosity = DotNetVerbosity.Minimal,
-            NoRestore = true
-        };
-
-        DotNetBuild(proj.ToString(), dotNetBuildSettings);
-
-        var strykerPath = Context.Tools.Resolve("Stryker.CLI.dll");
-        var mutationScore = XmlPeek(proj, "/Project/PropertyGroup/MutationScore/text()", new XmlPeekSettings { SuppressWarning = true });
-        var score = int.Parse(mutationScore);
-
-        Information($"Running mutation tests for '{proj}'. Test Project: '{testProj}'");
+Task("__RunLegacyMutationTests")
+    .Does((context) =>
+{
+    var oldDirectory = context.Environment.WorkingDirectory;
+    context.Environment.WorkingDirectory = MakeAbsolute(Directory("test"));
```

Comment:
Why do we need to change the working directory when we didn't before?

Reply:
Previously we surrounding the whole test block with this.
https://github.com/App-vNext/Polly/blob/a50d47230c4b6587f8257f91f47cf491ef39a019/build.cake#L176-L185

Just because now we seperate the test in two place so this shown as new code.

Reply:
I still don't quite follow - dotnet test should be being invoked from the repo root and the working directory _shouldn't_ matter because we specify `--project` as a full path?

Reply:
Yes you are correct. I got that directory changes removed.